R/testHelpers.R

```diff
@@ -0,0 +1,186 @@
+  # note this path has to match the out.dir argument in animint2THML...
+  testDir <- file.path(testPath, "animint-htmltest")
+  # if the htmltest directory exists, wipe clean, then create an empty folder
+  unlink(testDir, recursive = TRUE)
+  dir.create(testDir)
+  # start-up remote driver 
+  if (browserName == "phantomjs") {
+    message("Starting phantomjs binary. To shut it down, run: \n pJS$stop()")
+    pJS <<- RSelenium::phantom()
+  } else {
+    message("Starting selenium binary. To shut it down, run: \n",
+            "remDr$closeWindow() \n",
+            "remDr$closeServer()")
+    RSelenium::checkForServer(dir = system.file("bin", package = "RSelenium"))
+    selenium <- RSelenium::startServer()
+  }
+  # give an binaries a moment to start up
+  Sys.sleep(5)
+  remDr <<- RSelenium::remoteDriver(browserName = browserName, ...)
+  # give the backend a moment to start-up
+  Sys.sleep(4)
+  remDr$open(silent = TRUE)
+  Sys.sleep(2)
+  # if we navigate to localhost:%s/htmltest directly, some browsers will
+  # redirect to www.htmltest.com. A 'safer' approach is to navigate, then click.
+  remDr$navigate(sprintf("http://localhost:%s", port))
+  e <- remDr$findElement("xpath", "//a[@href='animint-htmltest/']")
+  e$clickElement()
+  invisible(TRUE)
+}
+
+#' Run animint tests
+#'
+#' Convenience function for running animint tests.
+#'
+#' @param dir character string with the path to animint's source code. Defaults to current directory
+#' @param filter If not NULL, only tests with file names matching
+#' this regular expression will be executed. Matching will take on the
+#' file name after it has been stripped of "test-" and ".r".
+#' @export
+#' @examples
+#'
+#' \dontrun{
+#' # run tests in test-rotate.R with Firefox
+#' tests_init("firefox")
+#' tests_run(filter = "rotate")
+#' # clean-up
+#' tests_exit()
+#' }
+#'
+
+tests_run <- function(dir = ".", filter = NULL) {
+  if (!"package:testthat" %in% search()) {
+    message("testthat must be loaded to run tests. Attempting to load for you...")
+    library("testthat")
+  }
+  testDir <- find_test_path(dir)
+  # functions that are reused across tests
+  source(file.path(testDir, "functions.R"))
+  # testthat::test_check assumes we are in path/to/animint/tests
+  old <- getwd()
+  on.exit(setwd(old), add = TRUE)
+  setwd(dirname(testDir))
+  # avoid weird errors if this function is called via testhat::check()
+  # https://github.com/hadley/testthat/issues/144
+  Sys.setenv("R_TESTS" = "")
+  test_check("animint", filter = filter)
+}
+
+#' Kill child process(es) that may have been initiated in animint testing
+#'
+#' Read process IDs from a file and kill those process(es)
+#' 
+#' @seealso \link{tests_run}
+#' @export
+tests_exit <- function() {
+  res <- stop_binary()
+  e <- try(readLines(pid_file(), warn = FALSE), silent = TRUE)
+  if (!inherits(e, "try-error")) {
+    pids <- as.integer(e)
+    res <- c(res, tools::pskill(pids))
+  }
```

Comment:
Yes. 

That approach will display an error message if there is an error, right? I think it makes sense to fail silently in this case.